tests/std/tests/P0528R3_cmpxchg_pad/test.cpp

```diff
@@ -0,0 +1,182 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include <atomic>
+#include <cassert>
+#include <cstddef>
+#include <cstring>
+
+struct X0 {
+    void operator&() const = delete;
+};
+
+
+struct X1 {
+    char x : 6;
+
+    void operator&() const = delete;
+
+    void set(char v) {
+        x = v;
+    }
+};
+
+struct X2 {
+    short x : 9;
+
+    void operator&() const = delete;
+
+    void set(char v) {
+        x = v;
+    }
+};
+
+#pragma pack(push, 1)
+struct X3 {
+    char x : 4;
+    char : 2;
+    char y : 1;
+    short z;
+
+    void operator&() const = delete;
+
+    void set(char v) {
+        x = v;
+        y = 0;
+        z = 0;
+    }
+};
+#pragma pack(pop)
+
+#pragma warning(push)
+#pragma warning(disable : 4324) // '%s': structure was padded due to alignment specifier
+struct alignas(4) X4 {
+    char x;
+
+    void operator&() const = delete;
+
+    void set(char v) {
+        x = v;
+    }
+};
+#pragma warning(pop)
+

```

Comment:
Can you add 5 and 9 for future ABI world where we round up to the next hardware atomic size coverage?

Reply:
Let it be 6 and 9. 
Types with odd size are boring, they are either `#pragma pack(1)` or only consist of byte pieces, 6 helps adding variation to how padding bits are obtained.